app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/widget/SwapWidget.kt

```diff
@@ -0,0 +1,99 @@
+package org.p2p.wallet.swap.ui.jupiter.main.widget
+
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.view.isVisible
+import android.content.Context
+import android.content.res.ColorStateList
+import android.text.InputType
+import android.util.AttributeSet
+import org.p2p.core.common.TextContainer
+import org.p2p.uikit.utils.drawable.shape.shapeRounded16dp
+import org.p2p.uikit.utils.drawable.shapeDrawable
+import org.p2p.uikit.utils.getColor
+import org.p2p.uikit.utils.inflateViewBinding
+import org.p2p.uikit.utils.text.TextViewCellModel
+import org.p2p.uikit.utils.text.bindOrGone
+import org.p2p.uikit.utils.toPx
+import org.p2p.wallet.R
+import org.p2p.wallet.databinding.WidgetSwapBinding
+
+class SwapWidget @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null
+) : ConstraintLayout(context, attrs) {
+
+    private val binding = inflateViewBinding<WidgetSwapBinding>()
+    private val initInputType: Int
+
+    init {
+        minHeight = 120.toPx()
+        background = shapeDrawable(shapeRounded16dp())
+        backgroundTintList = backgroundTint()
+        binding.textViewWidgetTitle.setTextColor(widgetTitleTint())
+        initInputType = binding.editTextAmount.inputType
+    }
+
+    fun bind(model: SwapWidgetModel) {
+        when (model) {
+            is SwapWidgetModel.Content -> bindContent(model)
+            is SwapWidgetModel.Loading -> bindLoading(model)
+        }
+    }
+
+    private fun bindLoading(model: SwapWidgetModel.Loading) {
```

Comment:
private fun bindLoading(model: SwapWidgetModel.Loading) = with(binding) {
and get rid of binding on every line